### PR TITLE
Support for many new model output types

### PR DESCRIPTION
**Pull request by abigailgold**
Including multi-label (output) models.
Note that not all model output types are supported in all modules and methods.

### CODE REVIEW

File: apt/utils/models/model.py [hunk 1]
```diff
@@ -115,16 +169,83 @@ def predict(self, x: Dataset, **kwargs) -> OUTPUT_DATA_ARRAY_TYPE:
...
 
-    @abstractmethod
     def score(self, test_data: Dataset, **kwargs):
         """
         Score the model using test data.
 
         :param test_data: Test data.
-        :type train_data: `Dataset`
+        :type test_data: `Dataset`
+        :param predictions: Model predictions to score. If provided, these will be used instead of calling the model's
```
Comment (andersonm-ibm) [inline apt/utils/models/model.py:178]: Why aren't all these defined in the API, but only in the method documentation?

File: tests/test_pytorch.py [hunk 2]
```diff
@@ -77,20 +155,208 @@ def test_nursery_pytorch_save_entire_model():
...
+
+    pred = model.predict(data)
+    assert (pred.shape[0] == x.shape[0])
+    score = model.score(data)
+    assert (0 < score <= 1.0)
+
+
+def test_pytorch_predictions_multi_label_cat():
+    # This kind of model requires special training and will not be supported using the 'fit' method.
+    class multi_label_cat_model(nn.Module):
```
Comment (andersonm-ibm) [inline tests/test_pytorch.py:255]: Class name should be MultiLabelCatModel

File: tests/test_pytorch.py [hunk 3]
```diff
@@ -77,20 +155,208 @@ def test_nursery_pytorch_save_entire_model():
...
+
+    pred = model.predict(test)
+    assert (pred.shape[0] == x_test.shape[0])
+
+    score = model.score(test, apply_non_linearity=expit)
+    assert (0 < score <= 1.0)
+
+
+def test_pytorch_predictions_multi_label_binary():
+    class multi_label_binary_model(nn.Module):
```
Comment (andersonm-ibm) [inline tests/test_pytorch.py:324]: Again, class names should normally use the CapWords convention, unless they are primarily used as a callable.

**Review by andersonm-ibm — CHANGES_REQUESTED** [review]
Some readability suggestions
